cmd/init.go

```diff
@@ -43,6 +44,112 @@ func (h *Handler) initNew(ctx context.Context, req *entity.CommandRequest) error
 	return nil
 }
 
+func (h *Handler) initFromTemplate(ctx context.Context, req *entity.CommandRequest) error {
+	// Prompt for a template to use
+
+	template, err := ui.PromptStarterTemplates()
+	if err != nil {
+		return err
+	}
+
+	// Select GitHub org
+
+	ui.StartSpinner(&ui.SpinnerCfg{Message: "Fetching GitHub scopes"})
+	scopes, err := h.ctrl.GetWritableGithubScopes(ctx)
+	if err != nil {
+		return err
+	}
+	if len(scopes) == 0 {
+		return errors2.NoGitHubScopesFound
+	}
+	ui.StopSpinner("")
+
+	org, err := ui.PromptGitHubScopes(scopes)
+	if err != nil {
+		return err
+	}
+
+	// Enter project name
+
+	name, err := ui.PromptProjectName()
+
+	// Prompt for env vars (if required)
+
+	variables := make(map[string]string)
+	if len(template.EnvVars) > 0 {
+		fmt.Printf("\n%s\n", ui.Bold("Environment Variables"))
+		for _, envVar := range template.EnvVars {
+			fmt.Printf("\n(%s)\n", ui.GrayText(envVar.Desc))
+			v, err := ui.PromptText(envVar.Name)
+			if err != nil {
+				return err
+			}
+			variables[name] = v
+		}
+		fmt.Print("\n")
+	}
+
+	// Create Railway project
+
+	ui.StartSpinner(&ui.SpinnerCfg{Message: "Creating project"})
+	creationResult, err := h.ctrl.CreateProjectFromTemplate(ctx, &entity.CreateProjectFromTemplateRequest{
+		Name:      name,
+		Org:       org,
+		Template:  template.Href,
+		IsPrivate: false,
+		Plugins:   template.Plugins,
+		Variables: variables,
+	})
+	if err != nil {
+		return err
+	}
+
+	project, err := h.ctrl.GetProject(ctx, creationResult.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Wait for workflow to complete
+
+	ui.StopSpinner("")
+	ui.StartSpinner(&ui.SpinnerCfg{
+		Message: "Deploying project",
+	})
+
+	for {
+		time.Sleep(2 * time.Second)
+		ws, err := h.ctrl.GetWorkflowStatus(ctx, creationResult.WorkflowID)
+		if err != nil {
+			return err
+		}
+		if ws.Status == "Error" {
+			ui.StopSpinner("")
+			return errors2.WorkflowFailed
+		}
+		if ws.Status == "Complete" {
```

Comment:
I'm unsure if we use these anywhere else, but we might want to hoist them into an enum and match against that vs doing direct string comp